backend/director/llm/xai.py

```
@@ -0,0 +1,181 @@
+                    "function": {
+                        "name": tool["name"],
+                        "description": tool["description"],
+                        "parameters": tool["parameters"],
+                    },
+                    "strict": True,
+                }
+            )
+        return formatted_tools
+
+    def chat_completions(
+        self, messages: list, tools: list = [], stop=None, response_format=None
```

Ruff (0.8.0)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)